src/main/java/de/tum/in/www1/artemis/service/connectors/ldap/LdapAuthenticationProvider.java

```diff
@@ -58,41 +61,101 @@ public Authentication authenticate(Authentication authentication) throws Authent
+            // without this code a second user would be created in Artemis which is not what we want (additionally this would fail because of unique constraints)
+            optionalUser = userRepository.findOneWithGroupsAndAuthoritiesByLogin(ldapUserDto.getLogin());
+        }
+        log.debug("Finished ldapUserService.findByLogin in {}", TimeLogUtil.formatDurationFrom(start));
         start = System.nanoTime();
 
-        // We create our own authorization and use the credentials of the user.
+        // Use the given password to compare it with the LDAP entry (i.e. check the authorization)
         byte[] passwordBytes = Utf8.encode(password);
         boolean passwordCorrect = ldapTemplate.compare(ldapUserDto.getUid().toString(), "userPassword", passwordBytes);
-        log.debug("Compare password with LDAP entry for user {} to validate login", username);
-        // this is the normal case, where the password is validated
+        log.debug("Compare password with LDAP entry for user {} to validate login", loginOrEmail);
         if (!passwordCorrect) {
             throw new BadCredentialsException("Wrong credentials");
         }
 
         log.debug("Finished ldapTemplate.compare password in {}", TimeLogUtil.formatDurationFrom(start));
 
-        return optionalUser.orElseGet(() -> {
-            User newUser = userCreationService.createUser(ldapUserDto.getUsername(), null, null, ldapUserDto.getFirstName(), ldapUserDto.getLastName(), ldapUserDto.getEmail(),
+        // update the user details from ldapUserDto (because they might have changed, e.g. when the user changes the name)
+        if (optionalUser.isPresent()) {
+            User user = optionalUser.get();
+            boolean saveNeeded = false;
+            if (!Objects.equals(user.getLogin(), ldapUserDto.getLogin())) {
+                user.setLogin(ldapUserDto.getLogin());
+                saveNeeded = true;
+            }
+            if (!Objects.equals(user.getFirstName(), ldapUserDto.getFirstName())) {
+                user.setFirstName(ldapUserDto.getFirstName());
+                saveNeeded = true;
+            }
+            if (!Objects.equals(user.getLastName(), ldapUserDto.getLastName())) {
+                user.setLastName(ldapUserDto.getLastName());
+                saveNeeded = true;
+            }
+            if (!Objects.equals(user.getEmail(), ldapUserDto.getEmail())) {
+                user.setEmail(ldapUserDto.getEmail());
+                saveNeeded = true;
+            }
+            if (!Objects.equals(user.getRegistrationNumber(), ldapUserDto.getRegistrationNumber())) {
+                user.setRegistrationNumber(ldapUserDto.getRegistrationNumber());
+                saveNeeded = true;
+            }
+            // only save the user in the database in case it has changed
+            if (saveNeeded) {
+                user = userRepository.save(user);
+            }
+            return user;
```

Comment:
This can be extracted:
1. One equals utils method to determine `saveNeeded`
2. One update utils method to update the user with the DTO values. (doesn't need equality checks as overriding with the same value doesn't hurt.

Reply:
I'm pretty sure Hibernate would try to save the user in the database because it subclasses `AbstractAuditingEntity`. And a save operation is relatively expensive, so I would avoid it